pkg/registry/build_registry/build_registry.go

```diff
@@ -167,10 +167,7 @@ func generateIEString(name string, elementid string, datatype string, enterprise
 	elementID, _ := strconv.ParseUint(elementid, 10, 16)
 	enterpriseID, _ := strconv.ParseUint(enterpriseid, 10, 16)
 	dataType := entities.IENameToType(datatype)
-	length := uint16(0)
-	if entities.IsValidDataType(dataType) {
-		length = entities.InfoElementLength[dataType]
-	}
+	length := entities.InfoElementLength[dataType]
```

Comment:
If some registry provides an info element with invalidDataType, we will still be adding the infoElement, right? This is not because of this fix. 
@zyiou  Should we allow registries to add info elements with invalidDataType? I feel we should throw an error when building the registry. What do you think?

Reply:
yes. we should not allow adding info elements with invalidDataType. Will fix it in another patch.